### ISSUE TITLE
mergeDataSets fails when blank value is used for merging

### ISSUE THREAD

**Issue by RomashkoVictor** [database-rider/database-rider#243]
To use mergeDatasets feature we should add **'empty.yml'** dataset to class level. In other case we will receive error:

```Caused by: java.lang.RuntimeException: No dataset loaded for name 
	at com.github.database.rider.core.dataset.DataSetExecutorImpl.loadDataSet(DataSetExecutorImpl.java:295)
	at com.github.database.rider.core.dataset.DataSetExecutorImpl.loadDataSets(DataSetExecutorImpl.java:305)
	at com.github.database.rider.core.dataset.DataSetExecutorImpl.createDataSet(DataSetExecutorImpl.java:149)
	... 74 more
```
It can be fixed simply by removing null and empty strings after merge.
```
public static DataSet mergeDataSetAnnotations(DataSet classLevelDataSet, DataSet methodLevelDataSet) {
        //only the lists are merged
        String[] value = joinArray(classLevelDataSet.value(), methodLevelDataSet.value());
        value = removeEmptyAndNullElements(value);
        ...
        return mergedDataSet;
    }
    public static String[] removeEmptyAndNullElements(String[] values) {
        List<String> result = new ArrayList<>(values.length);
        for(String value:values){
            if(value != null && !value.trim().isEmpty()) {
                result.add(value);
            }
        }
        return result.toArray(new String[0]);
    }
```

**Comment by cselagea** [comment]
I'm running into this issue as well.

I'm trying to test an insert, so the test method has `@DataSet(cleanBefore = true)`, but since there is a foreign key constraint, I need to insert some data into another table, for which I'm using a class-level annotation `@DataSet("datasets/price-files.yml")` because other test methods need this data as well.

After the merge, the `DataSet` value contains `"datasets/price-files.yml"` and `""`. The empty string leads to `java.lang.RuntimeException: No dataset loaded for name `.

**Comment by rmpestano** [comment]
Hi guys,

sorry but I'm a bit confused with this issue, do you have a full example?


> To use mergeDatasets feature we should add 'empty.yml' dataset to class level. In other case we will receive error:

 I can definitely use `empty.yml` in merge datasets, what may confuse is just the order of how datasets are processed.

The `test` level dataset is processed first and later is the class level:
<img width="1039" alt="Screenshot 2020-11-08 at 10 26 45" src="https://user-images.githubusercontent.com/1592273/98461559-64416c00-21ad-11eb-8b57-1bc1db3fbaa6.png">

Here I invert the order and then empty dataset is processed after users.yml so that's why the table is empty:
<img width="657" alt="Screenshot 2020-11-08 at 10 28 11" src="https://user-images.githubusercontent.com/1592273/98461574-83d89480-21ad-11eb-8fe1-870237cd9fbb.png">

Maybe if you give more details I can help you

**Comment by rmpestano** [comment]
> To use mergeDatasets feature we should add 'empty.yml' dataset to class level. In other case we will receive error:

Actually you're saying you have to use empty.yml to make it work, I also don't understand why...so please update the issue with complete examples.

Thanks!

**Comment by rmpestano** [comment]
Ok finally got it, you have `@DataSet(value="")`:

```
@RunWith(JUnit4.class)
@DBUnit(mergeDataSets = true)
@DataSet(value = "")
public class MergeDataSetsIt {

    @Rule
    public EntityManagerProvider emProvider = EntityManagerProvider.instance("rules-it");

    @Rule
    public DBUnitRule dbUnitRule = DBUnitRule.instance(emProvider.connection());

    @Test
    @DataSet(value = "yml/user.yml")
    public void shouldMergeDataSetsFromClassAndMethod() {
        List<User> users = em().createQuery("select u from User u").getResultList();
        assertThat(users).isNotNull().isNotEmpty().hasSize(4);  //2 users from user.yml plus 1 from  class level 'executeStatementsBefore' and 1 user from method level 'executeStatementsBefore'
    }

}

```

**Comment by rmpestano** [comment]
The fix will be available in `v1.17.0` to be released today, thank you guys.

**Comment by RomashkoVictor** [comment]
Great, thanks=)
